### ISSUE TITLE
Web interface search returns nothing, but typing tag into URL shows results

### ISSUE THREAD

**Issue by mudomau** [mastodon/mastodon#19944]
### Steps to reproduce the problem

1. Search term "perovskite" on Mastodon web interface (using mas.to server).
2. Yields no results.
3. Type url "mas.to/tags/perovskite"
4. A bunch of results turn up.

### Expected behaviour

The search should have yielded results.

### Actual behaviour

The search did not yield results.

### Detailed description

1. Search term "perovskite" on Mastodon web interface (using mas.to server).
2. Yields no results.
3. Type url "mas.to/tags/perovskite"
4. A bunch of results turn up.

### Specifications

Mastodon 4.0, just updated server mas.to, using edge browser, adblock, privacy badger, keepa, windows 11

**Comment by greggawatt** [comment]
Do you have a video of the bug?

**Comment by mudomau** [comment]
> Do you have a video of the bug?

No, but here's a post with a couple of screenshots showing the behavior - https://mas.to/@mzedp/109300113519385908

It's as described, type keyword into search bar, press enter and wait. It returns "Could not find anything for these search terms".

Typing the same keyword into the URL as mas.to/tags/keyword does return results.

**Comment by greggawatt** [comment]
I see, tested this on the instance I'm on mastodon.lol and it works fine, there must be some bug with mas.to, have you reached out to the admin trumpet@mas.to to see if they see the error in the logs? I don't see anything in the browser console.

**Comment by mudomau** [comment]
Thanks for the quick reply. I did message trumpet but that was just now, they might not have seen it yet. Let's give it a while.

**Comment by greggawatt** [comment]
also when testing "pero" works, "perov" doesn't

**Comment by leo60228** [comment]
This is generally expected, since to prevent harassment searching will only include posts you've interacted with. It might be possible to have hashtags always be searched, though?
